server/server.js

```diff
@@ -34,7 +34,7 @@ app.post('/register', async (req, res) => {
 			text: "Your secret is " + userSecret
 	}};
 
-	request.post('https://slack.com/api/chat.postMessage', data, function (error, response, body) {
+	request.post('https://slack.com/api/chat.postEphemeral', data, function (error, response, body) {
```

Comment:
I think you have to include the target users id to he data array as the api doc states these required: https://api.slack.com/methods/chat.postEphemeral
Also I would thing about using attachments (with pretext/text) instead of text only so the pretext can say: "Only visible to you" and the text states the secret.

Reply:
You are right, the userId is needed in the JSON sent into the request.
I think we don't need to explicitly say that the message is `seen only by you` as Slack already notify this beside the data/time (Check this example https://i.ibb.co/nwnz5q8/Screenshot-2019-06-26-at-21-19-14.png )